### PR TITLE
loadTimeDifference moved to base.exchange

### PR DESCRIPTION
**Pull request by samgermain**
```
% binance loadTimeDifference
2022-01-04T22:35:12.141Z
Node.js: v14.17.0
CCXT v1.66.23
binance.loadTimeDifference ()
624 ms
-153
```

### CODE REVIEW

File: python/ccxt/base/exchange.py [hunk 1]
```diff
@@ -436,6 +436,7 @@ def __init__(self, config={}):
 
         self.session = self.session if self.session or not self.synchronous else Session()
         self.logger = self.logger if self.logger else logging.getLogger(__name__)
+        self.has['loadTimeDifference'] = self.has['fetchTime']
```
Comment (samgermain) [inline python/ccxt/base/exchange.py:439]: This isn't working. When you print `exchange.has`, loadTimeDifference isn't a key in it

Comment (samgermain) [reply python/ccxt/base/exchange.py:439]: This will need to work to get this PR to compile properly

**Comment by kroitor** [comment]
I think, those `load` methods should not be a part of the `.has` schema. They're a part of the base class, or of a derived class, but not a part of the unified API, which should mostly be focused on the exchanges' functionality, not CCXT's functionality.

**Comment by samgermain** [comment]
> I think, those `load` methods should not be a part of the `.has` schema. They're a part of the base class, or of a derived class, but not a part of the unified API, which should mostly be focused on the exchanges' functionality, not CCXT's functionality. I general, not every method of the exchange is unified and not every method has to be listed in the `.has` property.

I'll take it out then, this is an issue for other methods too though, so I'm wondering if there's a way we could do this functionality so that I'm able to do it for other methods

File: js/latoken.js [hunk 2]
```diff
@@ -242,7 +236,7 @@ module.exports = class latoken extends Exchange {
         //         }
         //     ]
         //
-        if (this.safeValue (this.options, 'adjustForTimeDifference', true)) {
```
Comment (frosty00) [inline js/latoken.js:245]: this change is breaking latoken

Comment (frosty00) [reply js/latoken.js:245]: I don't think we should remove the guards from `this.options` that seems like a regression to me

Comment (samgermain) [reply js/latoken.js:245]: it's breaking anyway, changing this back won't fix it

**Comment by samgermain** [comment]
I don't know how to fix this

```
AILED latoken (Python 3 Async): (explain @ run-tests.js:190)

	latoken using proxy ``

	<class 'AttributeError'> 'latoken' object has no attribute 'load_time_difference' 

	

	  File "python/ccxt/test/test_async.py", line 579, in <module>

	    asyncio.run(main())

	

	  File "/opt/python/3.7.6/lib/python3.7/asyncio/runners.py", line 43, in run

	    return loop.run_until_complete(main)

	

	  File "/opt/python/3.7.6/lib/python3.7/asyncio/base_events.py", line 583, in run_until_complete

	    return future.result()

	

	  File "python/ccxt/test/test_async.py", line 566, in main

	    await try_all_proxies(exchange, proxies)

	

	  File "python/ccxt/test/test_async.py", line 504, in try_all_proxies

	    await load_exchange(exchange)

	

	  File "python/ccxt/test/test_async.py", line 388, in load_exchange

	    await exchange.load_markets()

	

	  File "/home/travis/build/ccxt/ccxt/python/ccxt/async_support/base/exchange.py", line 199, in load_markets

	    raise e

	

	  File "/home/travis/build/ccxt/ccxt/python/ccxt/async_support/base/exchange.py", line 195, in load_markets

	    result = await self.markets_loading

	

	  File "/home/travis/build/ccxt/ccxt/python/ccxt/async_support/base/exchange.py", line 185, in load_markets_helper

	    markets = await self.fetch_markets(params)

	

	  File "/home/travis/build/ccxt/ccxt/python/ccxt/async_support/latoken.py", line 248, in fetch_markets

	    await self.load_time_difference() (explain @ run-tests.js:193)

FAILED latoken (PHP Async): (explain @ run-tests.js:190)

	timed out (explain @ run-tests.js:193)

FAIL ["latoken"] (run-tests.js:271)
```

**Comment by kroitor** [comment]
@samgermain 

- in JS we use the `camelCase` notation for base methods
- in Py and PHP we use the `underscore_snake_notation`

Therefore it has to be `load_time_difference` in Py and PHP.
And we will have to add a mapping here: https://github.com/ccxt/ccxt/blob/master/build/transpile.js#L125

**Comment by samgermain** [comment]
> 

Does that mean these methods should also be added to transpile.js? 

https://github.com/ccxt/ccxt/pull/11158

**Comment by kroitor** [comment]
@samgermain 

> Does that mean these methods should also be added to transpile.js?
> 
> #11158

Yes, we have to hardcode base methods in transpile.js – that is until we add a better way of transpiling those methods.

**Comment by kroitor** [comment]
@samgermain if we are moving a method that is using an `async` implicit endpoint to the base class – we have to add it to python/ccxt/exchange.py in sync version and also to python/ccxt/async_support/exchange.py in the async-version. The same is true for PHP.

**Pull request merged**